### ISSUE TITLE
Make db service in docker-compose use non-root user

### ISSUE THREAD

**Issue by Creased** [CTFd/CTFd#1027]
**Environment**:

  - CTFd Version/Commit: b5632f9
  - Operating System: Docker v18.09.5
  - Web Browser and Version: not related

**What happened?**

```
Access denied for 'ctfd'@'%' to database 'ctfd'
```

**What did you expect to happen?**

Connect to CTFd database using ctfd user instead of using root.

**How to reproduce your issue**

```bash
docker-compose up -d
docker-compose logs -f
```

**Any associated stack traces or error logs**

```
ctfd_1   | [SQL: CREATE DATABASE ctfd CHARACTER SET = 'utf8mb4']
ctfd_1   | (Background on this error at: http://sqlalche.me/e/e3q8)
ctfd_1   | Waiting for db: to be ready
ctfd_1   | db is ready
ctfd_1   | Traceback (most recent call last):
ctfd_1   |   File "manage.py", line 8, in <module>
ctfd_1   |     app = create_app()
ctfd_1   |   File "/opt/CTFd/CTFd/__init__.py", line 137, in create_app
ctfd_1   |     url = create_database()
ctfd_1   |   File "/opt/CTFd/CTFd/utils/migrations/__init__.py", line 26, in create_database
ctfd_1   |     create_database_util(url, encoding="utf8mb4")
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy_utils/functions/database.py", line 577, in create_database
ctfd_1   |     result_proxy = engine.execute(text)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2166, in execute
ctfd_1   |     return connection.execute(statement, *multiparams, **params)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 982, in execute
ctfd_1   |     return self._execute_text(object_, multiparams, params)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1155, in _execute_text
ctfd_1   |     parameters,
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
ctfd_1   |     e, statement, parameters, cursor, context
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
ctfd_1   |     util.raise_from_cause(sqlalchemy_exception, exc_info)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 383, in raise_from_cause
ctfd_1   |     reraise(type(exception), exception, tb=exc_tb, cause=cause)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
ctfd_1   |     cursor, statement, parameters, context
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
ctfd_1   |     cursor.execute(statement, parameters)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/cursors.py", line 170, in execute
ctfd_1   |     result = self._query(query)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/cursors.py", line 328, in _query
ctfd_1   |     conn.query(q)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/connections.py", line 516, in query
ctfd_1   |     self._affected_rows = self._read_query_result(unbuffered=unbuffered)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/connections.py", line 727, in _read_query_result
ctfd_1   |     result.read()
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/connections.py", line 1066, in read
ctfd_1   |     first_packet = self.connection._read_packet()
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/connections.py", line 683, in _read_packet
ctfd_1   |     packet.check_error()
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/protocol.py", line 220, in check_error
ctfd_1   |     err.raise_mysql_exception(self._data)
ctfd_1   |   File "/usr/local/lib/python2.7/site-packages/pymysql/err.py", line 109, in raise_mysql_exception
ctfd_1   |     raise errorclass(errno, errval)
ctfd_1   | sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (1044, u"Access denied for user 'ctfd'@'%' to database 'ctfd'")
ctfd_1   | [SQL: CREATE DATABASE ctfd CHARACTER SET = 'utf8mb4']
ctfd_1   | (Background on this error at: http://sqlalche.me/e/e3q8)
```

**Comment by ColdHeat** [comment]
CTFd concerns itself with creating the database because it needs to set the utf8mb4 default for proper unicode handling. Can you do this with docker-compose without using root?

**Comment by ColdHeat** [comment]
I also don't really see this as an issue since this is within docker-compose.

**Comment by Creased** [comment]
Please refer to https://github.com/CTFd/CTFd/pull/1028#issuecomment-503572796.

**Comment by ColdHeat** [comment]
I decided the best solution for this now is to continue using the root user but begin migrating to an unprivileged user and allow the usage of the unprivileged user.

**Comment by ColdHeat** [comment]
Closed with #1457
